### PR TITLE
docs: Add new docs page for Gradle Plugin

### PR DESCRIPTION
**Pull request by marcospereira**
Depends on #341.

### CODE REVIEW

**Comment by marcospereira** [comment]
@casid @kelunik, this is now rebased on top of main branch, and ready for review/merge. :-)

File: docs/gradle-plugin.md [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+| `contentType`           | The content type of all templates. Either `Plain` or `Html`                       | `Html`                                                         |
+| `trimControlStructures` | Trims control structures, resulting in prettier output                            | `false`                                                        |
+| `htmlTags`              | Intercepts the given html tags during template compilation                        | None                                                           |
+| `htmlPolicyClass`       | An [`HtmlPolicy`][html-policy] used to check the parsed HTML at compile time      | None                                                           |
+| `htmlCommentsPreserved` | If HTML comments should be preserved in the output                                | `false`                                                        |
+| `binaryStaticContent`   | If to generate a [binary content](binary-rendering.md) resource for each template | `false`                                                        |
```
Comment (kelunik) [inline docs/gradle-plugin.md:67]:
```suggestion
| `htmlCommentsPreserved` | Whether HTML comments should be preserved in the output                                | `false`                                                        |
| `binaryStaticContent`   | Whether to generate a [binary content](binary-rendering.md) resource for each template | `false`                                                        |
```

Comment (kelunik) [reply docs/gradle-plugin.md:67]: Not sure about these, but `if to` sounds really strange.

File: docs/gradle-plugin.md [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+| `kotlinCompileArgs`     | Sets additional compiler arguments for `kte` templates                            | None                                                           |
+| `packageName`           | The package name, where template classes are generated to                         | [`Constants.PACKAGE_NAME_PRECOMPILED`][constants-package-name] |
+
+!!! info "About `htmlPolicyClass`"
+
+    The `htmlPolicyClass` will default to `gg.jte.html.OwaspHtmlPolicy` if the content type is `Html`.
+
+!!! warning "Clean all before precompiling"
+
+    The `precompileJte` task cleans the directory containing the compiled template classes every time it runs. In your local development environment, it may make more sense to use [hot reloading](hot-reloading.md).
```
Comment (kelunik) [inline docs/gradle-plugin.md:78]: Same comment here as in the other PR: The hint to hot reloading seems unrelated to the `Clean all before precompiling` warning.

File: docs/gradle-plugin.md [hunk 3]
```diff
@@ -0,0 +1,270 @@
...
+
+## `generateJte` task { #generate-task }
+
+This task generates all template classes in a sources directory. This only generates `.java`/`.kt` files, not `.class` files.
```
Comment (kelunik) [inline docs/gradle-plugin.md:82]:
```suggestion
This task generates all template classes in a sources directory. This only generates `.java`/`.kt` files, but does not compile them to `.class` files.
```

File: docs/gradle-plugin.md [hunk 4]
```diff
@@ -0,0 +1,270 @@
...
+| `targetDirectory`         | Destination directory to store generated templates.                               | `${project.build.directory}/generated-sources/jte`             |
+| `contentType`             | The content type of all templates. Either `Plain` or `Html`                       | `Html`                                                         |
+| `trimControlStructures`   | Trims control structures, resulting in prettier output                            | `false`                                                        |
+| `htmlTags`                | Intercepts the given html tags during template compilation                        | None                                                           |
+| `htmlCommentsPreserved`   | If HTML comments should be preserved in the output                                | `false`                                                        |
+| `binaryStaticContent`     | If to generate a [binary content](binary-rendering.md) resource for each template | `false`                                                        |
```
Comment (kelunik) [inline docs/gradle-plugin.md:130]: Same regarding `if` vs. `whether` applies here.

File: docs/maven-plugin.md [hunk 5]
```diff
@@ -188,7 +188,11 @@ The following parameters are available for this extension:
 
 #### `gg.jte.nativeimage.NativeResourcesExtension` { #native-resources-extension }
 
-See details about it in the [native-image documentation](pre-compiling.md#graalvm-native-image-support). This extension requires no parameters.
+!!! info "Version note"
+
+    Available since jte ^^**1.10.0**^^.
```
Comment (kelunik) [inline docs/maven-plugin.md:193]: What does `^^` do?

File: docs/gradle-plugin.md (hunk 1 above)
Comment (marcospereira) [reply docs/gradle-plugin.md:67]: English is not my first language, so it may indeed be strange. However, it is consistent with the javadoc comments. If we decide to adjust here, I would rather adjust the javadoc comments and the Maven plugin docs page.

File: docs/maven-plugin.md (hunk 5 above)
Comment (marcospereira) [reply docs/maven-plugin.md:193]: Underscore. It will be rendered as this:

Screenshot 2024-04-09 at 4 23 09 PM

If I remember correctly, I did it this way when migrating the docs to MkDocs.

File: docs/gradle-plugin.md (hunk 1 above)
Comment (kelunik) [reply docs/gradle-plugin.md:67]: I'm not a native speaker either.

File: docs/maven-plugin.md (hunk 5 above)
Comment (marcospereira) [reply docs/maven-plugin.md:193]: To be precise, it is an `insert` that is styled using an underline:

https://facelessuser.github.io/pymdown-extensions/extensions/caret/?h=ins#insert

Comment (kelunik) [reply docs/maven-plugin.md:193]: Thanks!

File: docs/gradle-plugin.md (hunk 1 above)
Comment (marcospereira) [reply docs/gradle-plugin.md:67]: Grammarly seems okay with this phrasing. I think we can leave it like this.
